lib/ansible/modules/network/vyos/vyos_static_routes.py

```diff
@@ -0,0 +1,718 @@
+    type: list
+    elements: dict
+    suboptions:
+      address_families:
+        description: A dictionary specifying the address family to which the static route(s) belong.
+        type: list
+        elements: dict
+        suboptions:
+          afi:
+            description:
+              - Specifies the type of route.
+            type: str
+            choices: ['ipv4', 'ipv6']
+            required: True
+          routes:
+            description: A ditionary that specify the static route configurations.
+            type: list
+            elements: dict
+            suboptions:
+              dest:
+                description:
+                  - An IPv4/v6 address in CIDR notation that specifies the destination network for the static route.
+                type: str
+                required: True
+              blackhole_config:
+                description:
+                  - Configured to silently discard packets.
+                type: dict
+                suboptions:
+                  type:
+                    description:
+                      - This is to configure only blackhole.
+                    type: str
+                  distance:
+                    description:
+                      - Distance for the route.
+                    type: int
+              next_hops:
+                description:
+                  - Next hops to the specified destination.
+                type: list
+                elements: dict
+                suboptions:
+                  forward_router_address:
+                    description:
+                      - The IP address of the next hop that can be used to reach the destination network.
+                    type: str
+                    required: True
+                  enabled:
+                    description:
+                      - Disable IPv4/v6 next-hop static route.
+                    type: bool
+                  admin_distance:
+                    description:
+                      - Distance value for the route.
+                    type: int
+                  interface:
+                    description:
+                      - Name of the outgoing interface.
+                    type: str
+  running_config:
+    description:
+      - The module, by default, will connect to the remote device and
+        retrieve the current running-config to use as a base for comparing
+        against the contents of source. There are times when it is not
+        desirable to have the task get the current running-config for
+        every task in a playbook.  The I(running_config) argument allows the
+        implementer to pass in the configuration to use as the base
+        config for comparison. This value of this option should be the
+        output received from device by executing command
+        C(show configuration commands | grep 'static route')
+    version_added: "2.10"
+    type: str
+  state:
```

Comment:
Since the states behave in the same manner, IMO, its redundant to describe it in every single RM. I have opened a docs request to add the description of the network resource module states in one place and then add a link to this doc in every RM.

https://github.com/ansible/ansible/issues/65598